itests/wdpost_dispute_test.go

```diff
@@ -285,36 +286,34 @@ func TestWindowPostDisputeFails(t *testing.T) {
 
 	// Wait until a proof has been submitted.
 	var targetDeadline uint64
+	var dInfo *dline.Info
 waitForProof:
 	for {
+		head, err := client.ChainHead(ctx)
+		require.NoError(t, err)
+
+		di, err := client.StateMinerProvingDeadline(ctx, maddr, head.Key())
+		require.NoError(t, err)
+
 		deadlines, err := client.StateMinerDeadlines(ctx, maddr, types.EmptyTSK)
 		require.NoError(t, err)
 		for dlIdx, dl := range deadlines {
 			nonEmpty, err := dl.PostSubmissions.IsEmpty()
```

Comment:
I wonder if maybe the error here is simply that this should be flipped - wait until _one_ deadline is non-empty and that should be enough, no need to compare deadline indexes or anything - the fact that this variable is `nonEmpty` suggests that was the plan, but it should be the inverse -- name it `isEmpty` and wait until it's false and then you're in an epoch where there's a post submission, wait until 5 epochs after the close of it and then dispute.